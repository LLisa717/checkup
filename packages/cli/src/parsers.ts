import { Parser, ParserName } from '@checkup/core';

let registeredParsers: Map<ParserName, Parser> = new Map<ParserName, Parser>();

/**
 *
 */
export function getRegisteredParsers() {
  return registeredParsers;
}

/**
 * @param parserName
 * @param parser
 */
export function registerParser(parserName: ParserName, parser: Parser) {
  registeredParsers.set(parserName, parser);
}
