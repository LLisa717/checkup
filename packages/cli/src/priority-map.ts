import { Priority } from '@checkup/core';

export default class PriorityMap<K, V> {
  maps: Map<any, Map<K, V>>;

  constructor() {
    this.maps = new Map<Priority, Map<K, V>>([
      [Priority.High, new Map<K, V>()],
      [Priority.Medium, new Map<K, V>()],
      [Priority.Low, new Map<K, V>()],
    ]);
  }

  get(priority: Priority, key: K) {
    let map = this.maps.get(priority);

    return map!.get(key);
  }

  set(priority: Priority, key: K, value: V) {
    let map = this.maps.get(priority);

    map!.set(key, value);
  }

  *interator() {
    for (let [, map] of this.maps) {
      for (let [, task] of map) {
        yield task;
      }
    }
  }

  [Symbol.iterator]() {
    return this.interator();
  }
}
