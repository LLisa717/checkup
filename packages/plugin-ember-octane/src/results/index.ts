export { default as OctaneMigrationStatusTaskResult } from './octane-migration-status-task-result';
