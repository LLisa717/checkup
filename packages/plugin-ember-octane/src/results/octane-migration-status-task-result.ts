import { CLIEngine } from 'eslint';
import { TaskName, TaskResult, ui } from '@checkup/core';

export default class OctaneMigrationStatusTaskResult implements TaskResult {
  constructor(public taskName: TaskName, public report: CLIEngine.LintReport) {}

  toConsole() {
    ui.styledHeader(this.taskName);
    ui.blankLine();
    ui.styledObject({
      'JS Error Count': this.report.errorCount,
    });
    ui.blankLine();
  }

  toJson() {
    return {};
  }
}
