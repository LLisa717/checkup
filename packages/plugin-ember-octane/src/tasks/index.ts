export { default as OctaneMigrationStatusTask } from './octane-migration-status-task';
