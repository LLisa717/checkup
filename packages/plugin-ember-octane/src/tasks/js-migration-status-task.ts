import {
  BaseTask,
  Category,
  Priority,
  TaskClassification,
  TaskName,
  TaskResult,
  ui,
} from '@checkup/core';

class JSMigrationStatusTaskResult implements TaskResult {
  toConsole() {
    ui.styledHeader(JSMigrationStatusTask.friendlyTaskName);
    ui.blankLine();
    ui.styledObject({
      lewis: 'miller',
    });
    ui.blankLine();
  }

  toJson() {
    return {};
  }
}

export default class JSMigrationStatusTask extends BaseTask {
  static taskName: TaskName = 'js-migration-status';
  static friendlyTaskName: TaskName = 'Ember Octane JS Migration Status';
  static taskClassification: TaskClassification = {
    category: Category.Core,
    priority: Priority.Medium,
  };

  async run(): Promise<TaskResult> {
    const result = new JSMigrationStatusTaskResult();

    return result;
  }
}
