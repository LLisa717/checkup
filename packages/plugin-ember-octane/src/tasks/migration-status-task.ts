import {
  BaseTask,
  Category,
  Parser,
  ParserName,
  Priority,
  TaskClassification,
  TaskName,
  TaskResult,
  ui,
} from '@checkup/core';

class MigrationStatusTaskResult implements TaskResult {
  toConsole() {
    ui.styledHeader(MigrationStatusTask.friendlyTaskName);
    ui.blankLine();
    ui.styledObject({
      lewis: 'miller',
    });
    ui.blankLine();
  }

  toJson() {
    return {};
  }
}

export default class MigrationStatusTask extends BaseTask {
  static taskName: TaskName = 'octane-migration-status';
  static friendlyTaskName: TaskName = 'Ember Octane Migration Status';
  static taskClassification: TaskClassification = {
    category: Category.Core,
    priority: Priority.Medium,
  };

  private parsers: Map<ParserName, Parser>;

  constructor(cliArguments: any, parsers: Map<ParserName, Parser>) {
    super(cliArguments);

    this.parsers = parsers;
  }

  async run(): Promise<TaskResult> {
    let esLintParser = this.parsers.get('eslint');
    esLintParser?.execute(['.']);
    let result = new MigrationStatusTaskResult();

    return result;
  }
}
