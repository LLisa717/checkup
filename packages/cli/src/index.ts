import * as path from 'path';

import { Command, flags } from '@oclif/command';
import { TaskResult, getConfig, getPackageJson, loadPlugins, ui } from '@checkup/core';
import { getRegisteredParsers, registerParser } from './parsers';

import TaskList from './task-list';

function mergeTaskResults(taskResults: TaskResult[]) {
  let mergedResults = {};

  taskResults.forEach(taskResult => {
    mergedResults = Object.assign(mergedResults, taskResult.toJson());
  });

  return mergedResults;
}

class Checkup extends Command {
  static description = 'A CLI that provides health check information about your project';

  static args = [
    {
      name: 'path',
      required: true,
      description: 'The path referring to the root directory that Checkup will run in',
      default: '.',
    },
  ];

  static flags = {
    version: flags.version({ char: 'v' }),
    help: flags.help({ char: 'h' }),
    force: flags.boolean({ char: 'f' }),
    silent: flags.boolean({ char: 's' }),
    json: flags.boolean(),
    task: flags.string({ char: 't' }),
  };

  async run() {
    let { args, flags } = this.parse(Checkup);
    // let registeredTasks: Map<TaskName, TaskConstructor>;
    let taskResults: TaskResult[];
    let tasksToBeRun: TaskList = new TaskList();

    try {
      let checkupConfig = await getConfig(args.path);
      let plugins = await loadPlugins(checkupConfig.plugins, args.path);
      this.config.plugins.push(...plugins);
    } catch (error) {
      this.error(error);
    }

    try {
      getPackageJson(args.path);
    } catch (error) {
      this.error(
        `The ${path.resolve(
          args.path
        )} directory found through the 'path' option does not contain a package.json file. You must run checkup in a directory with a package.json file.`,
        error
      );
    }

    await this.config.runHook('register-parsers', {
      registerParser,
    });

    await this.config.runHook('register-tasks', {
      cliArguments: args,
      cliFlags: flags,
      parsers: getRegisteredParsers(),
      tasks: tasksToBeRun,
    });

    // registeredTasks = getRegisteredTasks();

    if (flags.task !== undefined) {
      // let task = registeredTasks.get(flags.task);
      // if (task !== undefined) {
      //   tasksToBeRun.registerTask(task, args);
      // }
      taskResults = [await tasksToBeRun.runTask(flags.task)];
    } else {
      // tasksToBeRun.registerTasks([...registeredTasks.values()], args);
      taskResults = await tasksToBeRun.runTasks();
    }

    ui.action.start('Checking up on your project');

    if (!flags.silent) {
      if (flags.json) {
        ui.styledJSON(mergeTaskResults(taskResults));
      } else {
        taskResults.forEach(taskResult => taskResult.toConsole());
      }
    }

    ui.action.stop();
  }
}

export = Checkup;
