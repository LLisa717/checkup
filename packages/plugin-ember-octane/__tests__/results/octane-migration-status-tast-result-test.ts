import { EmberProject, stdout } from '@checkup/test-helpers';
import { OctaneMigrationStatusTask } from '../../src/tasks';
import { OctaneMigrationStatusTaskResult } from '../../src/results';

describe('octane-migration-status-task-result', () => {
  let project: EmberProject;

  beforeEach(function() {
    project = new EmberProject('checkup-app', '0.0.0');
  });

  afterEach(function() {
    project.dispose();
  });

  describe('console output', () => {
    test('output to console', async () => {
      project.files = Object.assign(project.files, {
        app: {
          components: {
            'foo-bar.js': `
              import Component from '@glimmer/component';

              export default class FooBarComponent extends Component {}
            `,
          },
        },
      });

      project.writeSync();
      let task = new OctaneMigrationStatusTask({ path: project.baseDir });

      await task.run();

      let { report } = task;
      let taskResult = new OctaneMigrationStatusTaskResult('sdsds', report);

      taskResult.toConsole();

      expect(stdout()).toMatchSnapshot();
    });
  });
});
