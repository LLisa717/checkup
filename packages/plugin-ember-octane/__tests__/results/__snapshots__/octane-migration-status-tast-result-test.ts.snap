// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`octane-migration-status-task-result console output output to console 1`] = `
"=== sdsds

JS Error Count: 0

"
`;
